### PR TITLE
feat: add spec compliance verification to trace perspective

### LINKED ISSUE #311: Enhance trace perspective for spec compliance
### Problem

The `trace` (correctness) reviewer currently evaluates code correctness in isolation — it checks logic, error handling, and control flow but doesn't verify against the acceptance criteria. Since ACs will be injected into the review prompt (see: spec-aware reviews issue), `trace` should explicitly verify each AC against the diff, making it spec-compliance-aware.

### Users

**Primary**: Cerberus `trace` reviewer — gains spec compliance as a primary review dimension.
**Secondary**: PR authors — get explicit AC-by-AC compliance feedback.

### Recommended Approach

Enhance the `trace` perspective agent prompt to include spec compliance in its Primary Focus section. When ACs are present, `trace` explicitly maps each AC to diff code paths and flags missing test coverage.

### Acceptance Criteria

- [ ] [test] Given ACs are present in the review prompt, when `trace` (correctness) reviewer runs, then it explicitly verifies each AC against the diff
- [ ] [behavioral] Given an AC has no corresponding test in the diff, when `trace` runs, then it flags this as a finding (severity: minor)

### Non-Goals

- Adding a new reviewer for spec compliance (enhance `trace` instead)
- Making `trace` the sole spec compliance mechanism (other layers also verify)

### Verification

```bash
# Trace prompt references acceptance criteria
grep -qi 'acceptance\|spec compliance\|AC' [trace-perspective-prompt-file]
```

---

### Approach

Modify the `trace`/correctness perspective agent prompt to add spec compliance verification to its primary review scope. Locate the prompt file via `opencode.json` config (likely `pi/agents/` or `.opencode/agents/`).

### Files to Modify

- Perspective agent prompt for `trace`/correctness — add AC verification to Primary Focus section

### Prompt Addition

Add to `trace` reviewer's Primary Focus:

```markdown
### Spec Compliance
When acceptance criteria are provided in the review context:
1. For each AC, trace the code path through the diff that satisfies it
2. Report SATISFIED if the diff clearly implements the AC
3. Report NOT_SATISFIED if the AC requirement is missing or contradicted
4. Report CANNOT_DETERMINE if the diff is ambiguous
5. Flag any AC that has no corresponding test in the diff (severity: minor)
```

### Risks & Mitigations

- Prompt inflation → spec compliance section is ~100 words, minimal impact
- `trace` may over-focus on ACs vs. general correctness → frame as "additional lens" not replacement

---

**Epic**: Swiss Cheese Verification
**Dependencies**: #310 (spec-aware reviews)
**Downstream**: #312 (AC compliance in verdict JSON)

### PR DESCRIPTION
**Pull request by moomooskycow**
## Why This Matters

The `trace` (correctness) reviewer evaluates code correctness in isolation but doesn't verify against acceptance criteria. With ACs already injected into the review context (#310), trace should explicitly map each AC to diff code paths — giving PR authors structured, per-AC compliance feedback instead of generic correctness checks.

Closes #311.

## Trade-offs / Risks

- **Prompt inflation:** +13 lines (~100 words) to a 264-line prompt. Minimal impact on token budget.
- **Over-focus risk:** Framed as "check when present" not "always check" — trace falls back to pure correctness when no ACs exist.

## Intent Reference

> When ACs are present, trace explicitly maps each AC to diff code paths and flags missing test coverage.

Source: [#311](https://github.com/misty-step/cerberus/issues/311)

## Changes

- `pi/agents/correctness.md` — Added "Spec Compliance" subsection to Primary scope with AC mapping rules (SATISFIED/NOT_SATISFIED/CANNOT_DETERMINE) and untested-AC flagging
- `pi/agents/correctness.md` — Added deconfliction entries (AC compliance → trace, general coverage gaps → proof)
- `tests/test_spec_compliance_guidance.py` — 3 regression tests asserting prompt content

## Alternatives Considered

- **New reviewer for spec compliance** — Rejected per issue; enhancing trace is lighter and avoids panel bloat
- **Do nothing** — ACs injected but ignored by trace; wastes the #310 investment

## Acceptance Criteria

- [x] Given ACs are present in the review prompt, when `trace` runs, then it explicitly verifies each AC against the diff
- [x] Given an AC has no corresponding test in the diff, when `trace` runs, then it flags this as a finding (severity: minor)

## Manual QA

```bash
grep -qi 'acceptance\|spec compliance\|AC' pi/agents/correctness.md
python3 -m pytest tests/test_spec_compliance_guidance.py -v
```

## What Changed

```mermaid
graph LR
    A[PR Diff] --> B[trace reviewer]
    B --> C[Correctness findings]
```

```mermaid
graph LR
    A[PR Diff] --> B[trace reviewer]
    B --> C[Correctness findings]
    B --> D[AC compliance mapping]
    D --> E[SATISFIED / NOT_SATISFIED / CANNOT_DETERMINE]
    D --> F["Untested AC findings (minor)"]
```

Trace now has a dual lens: correctness bugs AND spec compliance when ACs are available.

## Test Coverage

- `tests/test_spec_compliance_guidance.py` — 3 tests covering section presence, AC status labels, and untested-AC flagging
- Full suite: 1840 passed, 0 failures

## Merge Confidence

**High.** Prompt-only change with regression tests. No code logic modified. Full suite green.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Release Notes

* **Documentation**
  * Enhanced correctness guidance with acceptance criteria compliance procedures, including mapping across code changes and handling of untested items.

* **Tests**
  * Added regression tests validating acceptance criteria compliance documentation is present and complete.

### CODE REVIEW

File: pi/agents/correctness.md [hunk 1]
```diff
@@ -47,6 +47,19 @@ Find logic bugs, edge cases, and incorrect behavior introduced by this PR.
 - CAS/atomic retry loops with ABA or no-progress risks
 - State transitions that can become inconsistent
 
+### Spec Compliance (check when acceptance criteria are present)
+When acceptance criteria (ACs) are provided in the review context via `github_read`:
+1. For each AC, trace the code path through the diff that satisfies it.
+2. Report SATISFIED if the diff clearly implements the AC.
+3. Report NOT_SATISFIED if the AC requirement is missing or contradicted by the diff.
+4. Report CANNOT_DETERMINE if the diff is ambiguous or you lack enough context.
```
[the remark this answers is not in the document]
Comment (phrazzld) [reply pi/agents/correctness.md]: The AC status labels (SATISFIED/NOT_SATISFIED/CANNOT_DETERMINE) are prompt reasoning guidance, not new schema fields. The model manifests AC compliance through the existing findings mechanism:

- **NOT_SATISFIED** → finding (major/critical)
- **CANNOT_DETERMINE** → finding (minor/info) if worth flagging
- **Untested AC** → finding (minor, category: `untested-acceptance-criterion`)
- **SATISFIED** → no finding needed

No schema change required. The output contract is already expressive enough for this.

**Comment by phrazzld** [comment]
**CodeRabbit nitpick (pytest fixture):** Valid suggestion. However, for 3 tests reading a small file, the I/O savings are negligible and the indirection adds complexity without proportional benefit. This follows the existing pattern in `test_infra_prompt_guidance.py` where each test reads independently. Will reconsider if this file grows significantly.

**Codex P2 (schema compliance):** Addressed inline — AC status labels are reasoning guidance for the model, not new JSON schema fields. Compliance is expressed through the existing findings mechanism.
